src/hotspot/share/opto/c2_globals.hpp

```diff
@@ -73,6 +73,15 @@
   develop(bool, StressBailout, false,                                       \
           "Perform bailouts randomly at C2 failing() checks")               \
                                                                             \
+  product(bool, OptimizeReachabilityFences, true, DIAGNOSTIC,               \
+          "Optimize reachability fences")                                   \
+                                                                            \
+  product(bool, PreserveReachabilityFencesOnConstants, false, DIAGNOSTIC,   \
+          "Preserve reachability fences on constants")                      \
+                                                                            \
+  product(bool, StressReachabilityFences, false, DIAGNOSTIC,                \
+          "Randomly insert ReachabilityFence nodes")                        \
```

Comment:
Drive-by sniping: what about a hello-world test where you test out these flags?

Reply:
Good idea. Added one.